src/environmentd/tests/util.rs

```diff
@@ -275,8 +286,12 @@ pub fn start_server(config: Config) -> Result<Server, anyhow::Error> {
     let persist_clients = PersistClientCache::new(persist_cfg, &metrics_registry);
     let persist_clients = Arc::new(Mutex::new(persist_clients));
     let postgres_factory = StashFactory::new(&metrics_registry);
-    let secrets_controller = Arc::clone(&orchestrator);
-    let connection_context = ConnectionContext::for_tests(orchestrator.reader());
+    let secrets_controller = config.secrets_controller.unwrap_or_else(|| {
+        let sc = Arc::clone(&orchestrator);
+        let sc: Arc<dyn SecretsController> = sc;
+        sc
+    });
```

Comment:
This seems a little funny. What about this?

```diff
diff --git a/src/environmentd/tests/util.rs b/src/environmentd/tests/util.rs
index 90b92d8165..8d00715931 100644
--- a/src/environmentd/tests/util.rs
+++ b/src/environmentd/tests/util.rs
@@ -82,7 +82,6 @@ use std::time::Duration;
 use std::{env, thread};
 
 use anyhow::anyhow;
-use mz_orchestrator::Orchestrator;
 use once_cell::sync::Lazy;
 use postgres::error::DbError;
 use postgres::tls::{MakeTlsConnect, TlsConnect};
@@ -286,11 +285,9 @@ pub fn start_server(config: Config) -> Result<Server, anyhow::Error> {
     let persist_clients = PersistClientCache::new(persist_cfg, &metrics_registry);
     let persist_clients = Arc::new(Mutex::new(persist_clients));
     let postgres_factory = StashFactory::new(&metrics_registry);
-    let secrets_controller = config.secrets_controller.unwrap_or_else(|| {
-        let sc = Arc::clone(&orchestrator);
-        let sc: Arc<dyn SecretsController> = sc;
-        sc
-    });
+    let secrets_controller = config
+        .secrets_controller
+        .unwrap_or_else(|| Arc::<ProcessOrchestrator>::clone(&orchestrator));
     let connection_context = ConnectionContext::for_tests(secrets_controller.reader());
     let inner = runtime.block_on(mz_environmentd::serve(mz_environmentd::Config {
         adapter_stash_url,
@@ -349,10 +346,6 @@ pub fn start_server(config: Config) -> Result<Server, anyhow::Error> {
     Ok(server)
 }
 
-trait OrchestratorSecretsController: Orchestrator + SecretsController {}
-
-impl<T: Orchestrator + SecretsController> OrchestratorSecretsController for T {}
-
 pub struct Server {
     pub inner: mz_environmentd::Server,
     pub runtime: Arc<Runtime>,

```

Reply:
cc @guswynn ^ this works too